build/tasks/CreateFrameworkListFile.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace RepoTasks
+{
+    public class CreateFrameworkListFile : Task
+    {
+        /// <summary>
+        /// Files to extract basic information from and include in the list.
+        /// </summary>
+        [Required]
+        public ITaskItem[] Files { get; set; }
+
+        [Required]
+        public string TargetFile { get; set; }
+
+        /// <summary>
+        /// Extra attributes to place on the root node.
+        ///
+        /// %(Identity): Attribute name.
+        /// %(Value): Attribute value.
+        /// </summary>
+        public ITaskItem[] RootAttributes { get; set; }
+
+        public override bool Execute()
+        {
+            XAttribute[] rootAttributes = RootAttributes
+                ?.Select(item => new XAttribute(item.ItemSpec, item.GetMetadata("Value")))
+                .ToArray();
+
+            var frameworkManifest = new XElement("FileList", rootAttributes);
```

Comment:
Doesn't this blow up if `rootAttributes` (or `RootAttributes`) is `null`? Suggest you remove the `null`-conditional in the `rootAtrributes` declaration (which can then use `var`) and make `RootAttributes` `[Required]`.

Reply:
This is copied code from corefx. I think there is an overall goal of unifying these tasks in arcade (Can't find the issue, but it's there 😄 ).

IMO RootAttributes should be Required and we remove nullables.